target/linux/pistachio/base-files/etc/init.d/bt-csr

```diff
@@ -0,0 +1,201 @@
+#!/bin/sh /etc/rc.common
+#
+# Copyright 2016 by Imagination Technologies Limited and/or its affiliated group companies.
+#
+# All rights reserved.  No part of this software, either
+# material or conceptual may be copied or distributed,
+# transmitted, transcribed, stored in a retrieval system
+# or translated into any human or computer language in any
+# form by any means, electronic, mechanical, manual or
+# other-wise, or disclosed to the third parties without the
+# express written permission of Imagination Technologies
+# Limited, Home Park Estate, Kings Langley, Hertfordshire,
+# WD4 8LZ, U.K.
+
+START=22
+STOP=22
+
+. /lib/functions/mac.sh
+. /lib/functions/pistachio.sh
+. /lib/functions/system.sh
+
+BT_RST_N_GPIO=0
+MARDUK_BOARD="Marduk"
+
+board=$(pistachio_board_name)
+
+if [ "$board" = "$MARDUK_BOARD" ]
+then
+	BT_RST_N_GPIO=43
+else
+	echo "Unkown Platform"
+	exit 1
+fi
+
+setup_csr_gpio() {
+	# Set up BT_RST_N MFIO access
+	# check if the gpio has already been exported
+	{
+	if [ ! -d "/sys/class/gpio/gpio$BT_RST_N_GPIO" ];then
+		echo $BT_RST_N_GPIO > /sys/class/gpio/export
+	fi
+	} > /dev/null 2>&1
+	echo out > /sys/class/gpio/gpio$BT_RST_N_GPIO/direction
+}
+
+hide_csr_gpio() {
+        # Remove BT_RST_N MFIO access
+        # check if the gpio has already been exported
+        {
+        if [ -d "/sys/class/gpio/gpio$BT_RST_N_GPIO" ];then
+                echo $BT_RST_N_GPIO > /sys/class/gpio/unexport
+        fi
+        } > /dev/null 2>&1
+}
+
+
+n_reset_csr() {
+	setup_csr_gpio
+	echo 1 > /sys/class/gpio/gpio$BT_RST_N_GPIO/value
+	hide_csr_gpio
+}
+
+reset_csr() {
+	setup_csr_gpio
+	echo 0 > /sys/class/gpio/gpio$BT_RST_N_GPIO/value
+	hide_csr_gpio
+}
+
+
+bringup_csr() {
+local PRIMARY_MAC_ADDR=$(get_primary_macaddr)
```

Comment:
Why don't we move all bluetooth mac address related logic too into mac.sh and just call get_bluetooth_mac function here?